pkg/services/serviceaccounts/api/api.go

```diff
@@ -0,0 +1,47 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/grafana/grafana/pkg/api/response"
+	"github.com/grafana/grafana/pkg/api/routing"
+	"github.com/grafana/grafana/pkg/middleware"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/accesscontrol"
+	acmiddleware "github.com/grafana/grafana/pkg/services/accesscontrol/middleware"
+	"github.com/grafana/grafana/pkg/services/serviceaccounts"
+)
+
+type ServiceAccountsAPI struct {
+	service        serviceaccounts.Service
+	accesscontrol  accesscontrol.AccessControl
+	RouterRegister routing.RouteRegister
+}
+
+func NewServiceAccountsAPI(
+	service serviceaccounts.Service,
+	accesscontrol accesscontrol.AccessControl,
+	routerRegister routing.RouteRegister,
+) *ServiceAccountsAPI {
+	return &ServiceAccountsAPI{
+		service:        service,
+		accesscontrol:  accesscontrol,
+		RouterRegister: routerRegister,
+	}
+}
+
+func (api *ServiceAccountsAPI) RegisterAPIEndpoints() {
+	auth := acmiddleware.Middleware(api.accesscontrol)
```

Comment:
Suggesting that we don't register the endpoints if the service-accounts service is not active.  Otherwise you have to guard every single function in service accounts.